Plutarch/DataRepr.hs

```diff
@@ -112,15 +114,25 @@ pmatchDataRepr d handlers =
               handler
               $ go common (idx + 1) rest constr
 
-newtype PIsDataReprInstances a s = PIsDataReprInstances (a s)
+newtype PIsDataReprInstances a h s = PIsDataReprInstances (a s)
```

Comment:
Do we need this `h`?

Reply:
Yes. The `h` is used by the `PLift` instance [here](https://github.com/Plutonomicon/plutarch/pull/62/files#diff-ea6f5550d24dc825b81b16e44af9bcb35390ed76d1d2522473df83429b44ac6eR83) to associate with `PHaskellType`.

This is the also reason why `PBuiltinType` has a `h` in it [here](https://github.com/Plutonomicon/plutarch/pull/62/files#diff-ea6f5550d24dc825b81b16e44af9bcb35390ed76d1d2522473df83429b44ac6eR74). I wonder if I should rename `PBuiltinType`.

Reply:
And at the use site:

```haskell
 deriving (PMatch, PIsData, PLift) via (PIsDataReprInstances PScriptPurpose Ledger.ScriptPurpose)
```

^ Associates the Haskell type as `h`.

Reply:
Alternatively, we can leave `PIsDataReprInstances` untouched, and create a new newtype (analogous to `PBuiltinType`, but for non-builtin types):

```haskell
newtype PUserType p h s = PUserType (p s)

...
 deriving (PMatch, PIsData) via (PIsDataReprInstances PScriptPurpose)
 deriving (PLift) via (PUserType PScriptPurpose Ledger.ScriptPurpose)
...
```